clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientAuthenticator.java

```diff
@@ -201,8 +213,31 @@ public void authenticate() throws IOException {
                     // Fall through and start SASL authentication using the configured client mechanism
                 }
             case INITIAL:
-                sendSaslClientToken(new byte[0], true);
-                setSaslState(SaslState.INTERMEDIATE);
+                sendInitialTokenAndSetIntermediateState();
```

Comment:
Looking at the rest of the code, I think it will be good to set the state here like in the rest of the code. Perhaps:
```
sendIntialToken();
setSaslState(SaslState.INTERMEDIATE);
```

Reply:
Done